### DISCUSSION TITLE
Raw behavioral videos

### DISCUSSION THREAD

**Discussion by laurelrr** [NeurodataWithoutBorders/helpdesk#8] [Q&A] [answered]
Hello,

In the current data set I am converting, there is a social competition set up between 2 mice.  There are raw behavioral videos of the two mice, extracted position data for each mouse, neural data collected from each mouse.  I plan to structure my nwb file around a single subject, but I am not sure what is the best practice to apply for storing raw videos or if it makes sense to try and link up the two nwb files in some way, as they are part of the same session.  

Here are my specific questions:
1) How can I link up two nwb files as being part of the same session?

2) Is there something equivalent to the electrical series for videos that would be appropriate for storing under the "raw" under the acquisition set, e.g., as a time series of video frames while also capturing the video meta data?  

3) For storing the position data, I have multiple key points that are tracked for each subject.  What is the recommendation on structuring the Spatial series with multiple points?  I have been working off the matlab example for position (https://neurodatawithoutborders.github.io/matnwb/tutorials/html/ecephys.html#H_4D8E86F2)
spatial_series_ts = types.core.SpatialSeries( ...
    'data', position_data, ...
    'reference_frame', '(0,0) is bottom left corner', ...
    'timestamps', linspace(0, 100)/200)
I don't see how to fit multiple points in this structure.  Nor do I see how to fit multiple positions into the 'Position' structure in that example.  Any help is much appreciated!

Thank you!

**Comment by bendichter** [comment, marked as answer]
1. NWB has tools for externally linking two files together, but the simplest way to do this would simply be to give both NWB files the same `session_id`.
2. Yes, use [ImageSeries](https://pynwb.readthedocs.io/en/stable/pynwb.image.html?highlight=ImageSeries#pynwb.image.ImageSeries).
3. I would recommend having multiple [SpatialSeries](https://pynwb.readthedocs.io/en/stable/pynwb.behavior.html?highlight=spatialseries%20#pynwb.behavior.SpatialSeries) objects each named according to what joint it is tracking, and put them all in a single [Position](https://pynwb.readthedocs.io/en/stable/pynwb.behavior.html?highlight=spatialseries%20#pynwb.behavior.Position) object.

**Reply by laurelrr** [reply]
Thank you!  Is ImageSeries only implemented on the pynwb side?  I don't see help files for it in Matlab.

**Reply by bendichter** [reply]
You can find the MATLAB API documentation for ImageSeries [here](https://neurodatawithoutborders.github.io/matnwb/doc/+types/+core/ImageSeries.html). Unfortunately, we don't have any MATLAB tutorials yet that use this exact type.

**Comment by laurelrr** [comment]
Is this the correct way to add multiple SpatialSeries to a Position, just tack it on?
```
% create Position object and add SpatialSeries
Position = types.core.Position('SpatialSeries_1', spatial_series_ts,...
                                                 'SpatialSeries_2', spatial_series_ts);
```

**Comment by bendichter** [comment]
@laurelrr Yes, you can do that. You can also do:

```matlab
position = types.core.Position();
position.spatialseries.set('SpatialSeries_1', spatial_series_ts_1);
position.spatialseries.set('SpatialSeries_2', spatial_series_ts_2);
```
